details.js

```diff
@@ -0,0 +1,69 @@
+document.addEventListener("DOMContentLoaded", function() {
+    const countryName = new URLSearchParams(location.search).get('name');
```

Comment:
You have a lot of code inside your event listner - it will make is heavy and hard to maintain the code 

Also it may affect performance so I would extract the code into separate functions and simply call them in here